src/main/java/org/folio/rest/repository/AccountRepository.java

```diff
@@ -45,6 +48,14 @@ public AccountRepository(Context context, Map<String, String> headers) {
     return promise.future();
   }
 
+  public Future<Map<String, Account>> getAccountsById(List<String> accountIds) {
+    Promise<Map<String, Account>> promise = Promise.promise();
+    pgClient.getById(ACCOUNTS_TABLE, new JsonArray(accountIds), Account.class, promise);
+    return promise.future()
+      .map(accountsMap -> accountIds.stream()
```

Comment:
Both for the same reason. The result may not be complete - some of the accounts may be nonexistent. We need to preserve the initial list of IDs to be able to say later on,  which ones were fetched from the DB and which ones weren't. 
Your next question would probably be "why are we not using `Collectors.toMap` instead of a custom collector?" - again, for the same reason. Some of the values in the final map may be null and `Collectors.toMap` has a known bug in OpenJDK that causes NPE in such cases https://bugs.openjdk.java.net/browse/JDK-8148463.